inception-ui-search/src/main/java/de/tudarmstadt/ukp/inception/app/ui/search/sidebar/ExternalSearchAnnotationSidebar.java

```diff
@@ -0,0 +1,459 @@
+            searchStateModel.getObject();
+        if (searchState.getCurrentRepository() == null && repositories.size() > 0) {
+            searchState.setCurrentRepository(repositories.get(0));
+        }
+        if (searchState.getTargetQuery() == null) {
+            searchState.setTargetQuery(Model.of(""));
+        }
+
+        repositoriesModel = LoadableDetachableModel.of(() -> externalSearchService
+            .listDocumentRepositories(project));
+
+        mainContainer = new WebMarkupContainer("mainContainer");
+        mainContainer.setOutputMarkupId(true);
+        add(mainContainer);
+
+        DocumentRepositorySelectionForm projectSelectionForm = new DocumentRepositorySelectionForm(
+            "repositorySelectionForm");
+        mainContainer.add(projectSelectionForm);
+
+        SearchForm searchForm = new SearchForm("searchForm");
+        add(searchForm);
+        mainContainer.add(searchForm);
+
+        List<IColumn<ExternalSearchResult, String>> columns = new ArrayList<>();
+
+        columns.add(new AbstractColumn<ExternalSearchResult, String>(new Model<>("Results"))
+        {
+            @Override
+            public void populateItem(Item<ICellPopulator<ExternalSearchResult>> cellItem,
+                String componentId, IModel<ExternalSearchResult> model)
+            {
+                @SuppressWarnings("rawtypes")
+                Item rowItem = cellItem.findParent( Item.class );
+                int rowIndex = rowItem.getIndex();
+                ResultRowView rowView = new ResultRowView(componentId, rowIndex + 1, model);
+                cellItem.add(rowView);
+            }
+        });
+
+        if (searchState.getDataProvider() == null)
+        {
+            searchState.setDataProvider(new ExternalResultDataProvider(externalSearchService,
+                userRepository.getCurrentUser(),
+                searchStateModel.getObject().getCurrentRepository(),
+                "merck"));
+        }
+
+        dataTableContainer = new WebMarkupContainer("dataTableContainer");
+        dataTableContainer.setOutputMarkupId(true);
+        mainContainer.add(dataTableContainer);
+
+        DataTable<ExternalSearchResult, String> resultTable = new DefaultDataTable<>("resultsTable",
+            columns, searchState.getDataProvider(), 8);
+        resultTable.setCurrentPage(searchState.getCurrentPage());
+        dataTableContainer.add(resultTable);
+
+    }
+
+    @Override protected void onDetach()
+    {
+        // TODO is it possible to know page change event and alter the state there?
+        DataTable<ExternalSearchResult, String> resultTable =
+            (DataTable<ExternalSearchResult, String>) dataTableContainer.get("resultsTable");
+        searchStateModel.getObject().setCurrentPage(resultTable.getCurrentPage());
+        super.onDetach();
+    }
+
+    private void actionImportDocument(ExternalSearchResult aResult)
+    {
+        String documentTitle = aResult.getDocumentTitle();
```

Comment:
Yes, and a lot of other code here are the same as `SearchPage.java` of `inception-ui-external-search` module.

Reply:
Let's try to refactor the shared code as much as possible (so that e.g. https://github.com/inception-project/inception/issues/824 only needs to be fixed in one place).  You could have some sort of `DocumentImporter` object to hold the shared logic, for example.

Reply:
Actually this particular function `actionImportDocument` is a bit different from the one in `SearchPage.java`.

Reply:
True, but the part which uses the `AjaxRequestTarget` is easy to pull out

Reply:
Moved shared code to `DocumentImporter` class.